### ISSUE TITLE
Python exception when literalinclude lines that are beyond a file's bounds...

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#675]
When you use a '.. literalinclude::' which specifies lines to be included, that
are beyond the range of the actual file, a python 'IndexError: list index out
of range' exception is thrown. Apparently, sphinx doesn't check to see that the
requested lines are within the bounds of the file. Here's an example:

Assume 'foo.txt' only contains 3 lines. They could be anything. Here's the
sphinx markup to include lines 10-20 (lines that don't exist):

```
.. literalinclude:: foo.txt
    :lines: 10-20
```

When we build, such as 'make html', Python throws an exception:

```
Exception occurred:                                                             
  File "/usr/local/lib/python2.6/site-packages/Sphinx-1.0.7-py2.6.egg/sphinx/dirrctives/code.py", line 155, in run                                              
    lines = [lines[i] for i in linelist]                                        
IndexError: list index out of range                                             
The full traceback has been saved in /tmp/sphinx-err-mm9Uje.log, if you want to 
 eport the issue to the developers.                                             
Please also report this if it was a user error, so that a better error message ccn be provided next time.                                                       
Either send bugs to the mailing list at <http://groups.google.com/group/sphinx-ddv/>,                                                                           
or report them in the tracker at <https://bitbucket.org/birkenfeld/sphinx/issues//. Thanks!
```

I discovered this today by accidently including the wrong file in one of my
documentation projects, a file which wasn't long enough to include the
requested lines.

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/675
- Originally reported by: Anonymous
- Originally created at: 2011-04-14T07:03:32.747

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2011-05-15 11:52:59+00:00_

Closes #675: Fix IndexErrors when including nonexisting lines with :rst:dir:`literalinclude`.

→ <<cset 02a3b5bfdfe14ffb348d2abcc33c2da9df51c1ce>>

**Comment by shimizukawa** [comment]
_From Georg Brandl on 2014-10-18 07:32:18+00:00_

Removing version: 1.0.7 (automated comment)
